packages/smooth_app/lib/pages/product/simple_input_widget.dart

```diff
@@ -50,87 +50,97 @@ class _SimpleInputWidgetState extends State<SimpleInputWidget> {
           ),
         ),
         ExplanationWidget(widget.helper.getAddExplanations(appLocalizations)),
-        Row(
-          mainAxisAlignment: MainAxisAlignment.spaceBetween,
-          crossAxisAlignment: CrossAxisAlignment.center,
-          children: <Widget>[
-            Flexible(
-              flex: 1,
-              child: Padding(
-                padding: const EdgeInsets.only(left: LARGE_SPACE),
-                child: RawAutocomplete<String>(
-                  key: _autocompleteKey,
-                  focusNode: _focusNode,
-                  textEditingController: widget.controller,
-                  optionsBuilder: (final TextEditingValue value) async {
-                    final List<String> result = <String>[];
-                    final String input = value.text.trim();
-                    if (input.isEmpty) {
-                      return result;
-                    }
-                    final TagType? tagType = widget.helper.getTagType();
-                    if (tagType == null) {
-                      return result;
-                    }
-                    // TODO(monsieurtanuki): ask off-dart to return Strings instead of dynamic?
-                    final List<dynamic> data =
-                        await OpenFoodAPIClient.getAutocompletedSuggestions(
-                      tagType,
-                      language: ProductQuery.getLanguage()!,
-                      limit: 1000000, // lower max count on the server anyway
-                      input: value.text.trim(),
-                    );
-                    for (final dynamic item in data) {
-                      result.add(item.toString());
-                    }
-                    result.sort();
-                    return result;
-                  },
-                  fieldViewBuilder: (BuildContext context,
-                          TextEditingController textEditingController,
-                          FocusNode focusNode,
-                          VoidCallback onFieldSubmitted) =>
-                      TextField(
-                    controller: textEditingController,
-                    decoration: InputDecoration(
-                      filled: true,
-                      border: const OutlineInputBorder(
-                        borderRadius: CIRCULAR_BORDER_RADIUS,
-                        borderSide: BorderSide.none,
+        LayoutBuilder(
+          builder: (_, BoxConstraints constraints) {
+            return Row(
```

Comment:
I find reading this a bit difficult. What about moving part of it to a fun?

Reply:
> I find reading this a bit difficult. What about moving part of it to a fun?

@VaiTon In flutter we call that `Widget`s ;)
That said, I must say that code is not fun to read as there are almost no changes and the only difference is that the `LayoutBuilder` shifts the whole code. As we use flutter and automatic format there's not much we can do about it.
Perhaps this time move that `Row` into an ad-hoc `Widget`: that won't solve @VaiTon's problem now but at least next time any change would be easier to review. No a strong requirement from my side, though.

Reply:
I have split the child of the AutocompleteList to a dedicated Widget.
However for the simple_input_widget file, there is too much work to do, and is thus outside the goal of this PR.

Please open an issue if necessary instead 👍